step-maven-plugin/step-commons-maven-plugin/src/main/java/step/plugins/maven/AbstractRunDeployedExecutionBundleMojo.java

```diff
@@ -0,0 +1,52 @@
+package step.plugins.maven;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import step.core.repositories.RepositoryObjectReference;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractRunDeployedExecutionBundleMojo extends AbstractRunExecutionBundleMojo {
+
+	@Parameter(property = "step-run-exec-bundle.step-maven-settings", required = false)
+	private String stepMavenSettings;
+
+	public AbstractRunDeployedExecutionBundleMojo() {
+	}
+
+	public void execute() throws MojoExecutionException {
+		getLog().info("Run step execution for deployed module " + getBuildFinalName() + " (version=" + getProjectVersion() + ")");
+
+		// empty context here - we just call the execution client according to the  plugin parameters
+		executeBundleOnStep(new HashMap<>());
+	}
+
+	@Override
+	protected RepositoryObjectReference prepareExecutionRepositoryObject(Map<String, Object> executionContext) {
+		return new RepositoryObjectReference("Artifact", prepareRepositoryParameters());
+	}
+
+	private HashMap<String, String> prepareRepositoryParameters() {
+		HashMap<String, String> repoParams = new HashMap<>();
+		repoParams.put("groupId", getGroupId());
```

Comment:
These constants are declared in ArtifactRepository (which we don't want to import here). If we have a suitable common project we could move the constants there